thrift/thrift.bzl

```diff
@@ -54,19 +55,34 @@ def _valid_thrift_deps(targets):
     if not hasattr(target, "thrift"):
       fail("thrift_library can only depend on thrift_library", target)
 
-# Some notes on the raison d'etre of thrift_library vs. scrooge_scala_library.
-# The idea is to be able to separate concerns -- thrift_library is concerned
-# with the ownership of thrift files, and organizing them into targets. It is
-# not concerned with how the process of converting thrifts into sources. Thus,
-# the scrooge_scala_library is what handles the specifics of code generation...
-# this is useful because it means that if there are different code generation
-# targets, we don't need to have a whole separate tree of targets organizing
-# the thrifts.
+# Some notes on the raison d'etre of thrift_library vs. code gen specific
+# targets. The idea is to be able to separate concerns -- thrift_library is
+# concerned purely with the ownership and organization of thrift files. It
+# is not concerned with what to do with them. Thus, the code gen specific
+# targets  will take the graph of thrift_libraries and use them to generate
+# code. This organization is useful because it means that if there are
+# different code generation targets, we don't need to have a whole separate
+# tree of targets organizing the thrifts per code gen paradigm.
 thrift_library = rule(
   implementation = _thrift_library_impl,
   attrs = {
       "srcs": attr.label_list(allow_files=_thrift_filetype),
       "deps": attr.label_list(),
+      #TODO this is not necessarily the best way to do this... the goal
+      # is that we want thrifts to be able to be imported via an absolute
+      # path. But the thrift files have no clue what part of their path
+      # should serve as the base for the import... for example, if a file is
+      # in src/main/thrift/com/hello/World.thrift, if something depends on that
+      # via "include 'com/hello/World.thrift'", there is no way to know what
+      # path that should be relative to. One option is to just search for anything
+      # that matches that, but that could create correctness issues if there are more
+      # than one in different parts of the tree. Another option is to take an argument
+      # that references namespace, and base the tree off of that. The downside
+      # to that is that thrift_library then gets enmeshed in the details of code
+      # generation. This could also be something punted to scrooge_scala_library
+      # or whatever, but I think that we should make it such that the archive
+      # created by this is created in such a way that absolute imports work...
+      "absolute_prefix": attr.string(default='', mandatory=False),
       "_jar": attr.label(executable=True, default=Label("@bazel_tools//tools/jdk:jar"), single_file=True, allow_files=True),
       "_jdk": attr.label(default=Label("//tools/defaults:jdk"), allow_files=True),
```

Comment:
I don't think we actually need this. We are just using `jar` here, not `javac` or the libraries or anything, right? So, if `jar` needs `jdk`, I guess it declares that.